src/neuroconv/datainterfaces/behavior/video/internalvideointerface.py

```diff
@@ -0,0 +1,361 @@
+
+    def get_timestamps(self, stub_test: bool = False) -> np.ndarray:
+        """
+        Retrieve the timestamps for the data in this interface.
+
+        Parameters
+        ----------
+        stub_test : bool, default: False
+            If True, limits the number of frames to scan through to 10.
+
+        Returns
+        -------
+        timestamps : numpy.ndarray
+            The timestamps for the data stream.
+        """
+        if self._timestamps is not None:
+            return self._timestamps
+        else:
+            return self.get_original_timestamps(stub_test=stub_test)
+
+    def set_aligned_timestamps(self, aligned_timestamps: np.ndarray):
+        """
+        Replace all timestamps for this interface with those aligned to the common session start time.
+
+        Must be in units seconds relative to the common 'session_start_time'.
+
+        Parameters
+        ----------
+        aligned_timestamps : numpy.ndarray
+            The synchronized timestamps for data in this interface.
+        """
+        assert (
+            self._starting_time is None
+        ), "If setting both timestamps and starting times, please set the timestamps first so they can be shifted by the starting times."
+        self._timestamps = aligned_timestamps
+
+    def set_aligned_starting_time(self, aligned_starting_time: float, stub_test: bool = False):
+        """
+        Align all starting times for all videos in this interface relative to the common session start time.
+
+        Must be in units seconds relative to the common 'session_start_time'.
+
+        Parameters
+        ----------
+        aligned_starting_time : float
+            The common starting time for all segments of temporal data in this interface.
+        stub_test : bool, default: False
+            If timestamps have not been set to this interface, it will attempt to retrieve them
+            using the `.get_original_timestamps` method, which scans through each video;
+            a process which can take some time to complete.
+
+            To limit that scan to a small number of frames, set `stub_test=True`.
+        """
+        timing_type = self.get_timing_type()
+        if timing_type == "timestamps":
+            aligned_timestamps = self.get_timestamps(stub_test=stub_test) + aligned_starting_time
+            self.set_aligned_timestamps(aligned_timestamps=aligned_timestamps)
+        elif timing_type == "starting_time and rate":
+            self._starting_time = aligned_starting_time
+
+    def align_by_interpolation(self, unaligned_timestamps: np.ndarray, aligned_timestamps: np.ndarray):
+        raise NotImplementedError("The `align_by_interpolation` method has not been developed for this interface yet.")
+
+    def add_to_nwbfile(
+        self,
+        nwbfile: NWBFile,
+        metadata: Optional[dict] = None,
+        stub_test: bool = False,
+        chunk_data: bool = True,
```

Comment:
should we choose a better name for this one so buffered writing is not confused with chunking?

Reply:
Done!